Makefile.am

```diff
@@ -0,0 +1,144 @@
+	$(SED) "s:python /usr:python $(PREFIX):g" < dpkgfiles/armoryoffline.desktop > $(DESTDIR)$(datarootdir)/applications/armoryoffline.desktop
+	$(SED) "s:python /usr:python $(PREFIX):g" < dpkgfiles/armorytestnet.desktop > $(DESTDIR)$(datarootdir)/applications/armorytestnet.desktop
+
+install-exec-local:
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/extras
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/bitcoinrpc_jsonrpc
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/txjsonrpc
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/txjsonrpc/web
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/ui
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/pytest
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/BitTornado/BT1
+	$(MKDIR_P) $(DESTDIR)$(libdir)/armory/urllib3
+	$(MKDIR_P) $(DESTDIR)$(bindir)
+	$(SED) "s: /usr/lib: $(libdir):g" < dpkgfiles/armory > $(DESTDIR)$(bindir)/armory
+	chmod +x $(DESTDIR)$(bindir)/armory
+	cp *.py *.so README.md $(DESTDIR)$(libdir)/armory/
+	rsync -rupE armoryengine $(DESTDIR)$(libdir)/armory/
+	cp extras/*.py $(DESTDIR)$(libdir)/armory/extras
+	cp bitcoinrpc_jsonrpc/*.py $(DESTDIR)$(libdir)/armory/bitcoinrpc_jsonrpc
+	cp -R txjsonrpc/*.py $(DESTDIR)$(libdir)/armory/txjsonrpc
+	cp -R txjsonrpc/web/*.py $(DESTDIR)$(libdir)/armory/txjsonrpc/web
+	cp ui/*.py $(DESTDIR)$(libdir)/armory/ui
+	cp pytest/*.py $(DESTDIR)$(libdir)/armory/pytest
+	cp -R urllib3/* $(DESTDIR)$(libdir)/armory/urllib3
+	cp BitTornado/*.py $(DESTDIR)$(libdir)/armory/BitTornado
+	cp BitTornado/BT1/*.py $(DESTDIR)$(libdir)/armory/BitTornado/BT1
+	cp default_bootstrap.torrent $(DESTDIR)$(libdir)/armory
+
+all-test-tools: all
+	$(MAKE) -C cppForSwig/gtest
+
+gtest: all-test-tools
+	(cd cppForSwig/gtest && ./CppBlockUtilsTests)
+
+pytest: all
+	python -m unittest discover
+
+test: gtest pytest
+
+mo: messages
+	$(MAKE) -C po mo
+
+messages:
+	xgettext -o po/messages.pot --from-code=UTF-8 -L Python -ktr *.py armoryengine/*.py ui/*.py
+
+if TARGET_LINUX
+if BUILD_LINUX
+deb: all
+	@while [ -z "$$AMD64_CHROOT" ]; do \
+		read -r -p "Name of amd64 chroot: " AMD64_CHROOT; \
+	done && \
+	while [ -z "$$I386_CHROOT" ]; do \
+		read -r -p "Name of i386 chroot: " I386_CHROOT; \
+	done && \
+	python dpkgfiles/make_deb_package.py $$AMD64_CHROOT 64 1 1 0 && \
+	python dpkgfiles/make_deb_package.py $$I386_CHROOT 32 1 1 0
+endif
+endif
+
+if TARGET_DARWIN
+if BUILD_LINUX
+QMAKE=qmake-qt4
+else
+if BUILD_DARWIN
+QMAKE=qmake -spec macx-g++
+endif
+endif
+objc:
+	$(SIP) -w -x VendorID -t WS_MACX -t Qt_4_8_6 -x Py_v3 -B Qt_5_0_0 -o \
+		-P -g -c osxbuild/objc_armory/ $(SIP_CPPFLAGS) \
+		osxbuild/objc_armory/ArmoryMac.sip
+	$(QMAKE) osxbuild/objc_armory/ArmoryMac.pro -o osxbuild/objc_armory/
+	$(MAKE) -C osxbuild/objc_armory/ CC="$(CC)" CXX="$(CXX)" LINK="$(LINK)"
+
+if USE_GITIAN
+Armory.app: Armory-tmp-gitian
+	rm -rf osxbuild/$@
+	cp -R osxbuild/Armory-tmp-gitian osxbuild/$@
+else
+Armory.app: Armory-tmp
+	rm -rf osxbuild/$@
+	cp -R osxbuild/Armory-tmp osxbuild/$@
+endif
+
+Armory-tmp: all osxbuild/install-py3 osxbuild/install-qt5
```

Comment:
In my most recent commit, I fixed the OS X rules to have proper targets as best I could, as determined by the expected output of the rules. The osxbuild/Armory-tmp* rules still use recursive cp on a directory instead of listing out files, but I think that is okay. Files are copied from osxbuild/install-{py3,qt5}, which are generated by a script that clears out the directories each time the script runs. So osxbuild/install-{py3,qt5} act as a temporary staging area for the Python 3 and Qt 5 installs and therefore, I wouldn't expect it to end up with extra files, because the only opportunity for extra files to be added such that it affects the rest of the build is after the osxbuild/install-{py3,qt5} runs, but before the osxbuild/Armory.app runs.